docs/platforms/java/common/enriching-events/scopes/index.mdx

```diff
@@ -1,46 +1,109 @@
+
+A scope is generally valid inside of a callback or an execution context. This means that multiple parts of your application may have different scopes active at the same time. For instance, a web server might handle multiple requests at the same time, and each request may have different scope data to apply to its events.
+
+When you call a global function such as `Sentry.captureException`, internally Sentry discovers the active scopes and uses them when capturing the event.
+
+## Different Kinds of Scopes
+
+The Sentry SDK has three different kinds of scopes:
+
+- [Global scope](#global-scope)
+- [Isolation scope](#isolation-scope)
+- [Current scope](#current-scope)
 
-The default SDK integrations will push and pop scopes intelligently. For
-instance web framework integrations will create and destroy scopes around your
-routes or controllers.
+### Global Scope
 
-## How the Scope and Hub Work
+The global scope is applied to _all_ events, no matter where they originate. You can use it to store data that should apply to all events, such as environmental information.
 
-As you start using an SDK, a scope and hub are automatically created for you out
-of the box. It's unlikely that you'll interact with the hub directly unless you're
-writing an integration or you want to create or destroy scopes. Scopes, on the
-other hand are more user facing. You can call <PlatformIdentifier name="configure-scope" /> at any point in
-time to modify data stored on the scope. This is useful for doing things like
-[modifying the context](../context/).
+You can access the global scope via `Sentry.getGlobalScope()` or `Sentry.configureScope(ScopeType.GLOBAL, globalScope -> { ... })`.
 
-When you call a global function such as <PlatformIdentifier name="capture-event" /> internally Sentry
-discovers the current hub and asks it to capture an event. Internally the hub will
-then merge the event with the topmost scope's data.
+### Isolation Scope
+
+The isolation scope is used to isolate events from each other. For example, each request in a web server might get its own isolation scope, so that events from one request don't interfere with events from another request. In most cases, you'll want to put data that should be applied to your events on the isolation scope - which is also why all `Sentry.setXXX` methods, like `Sentry.setTag()`, will write data onto the currently active isolation scope. A classic example for data that belongs on the isolation scope is a user - each request may have a different user, so you want to make sure that the user is set on the isolation scope.
+
+You can modify the isolation scope via `Sentry.configureScope(ScopeType.ISOLATION, isolationScope -> { ... })`, but usually you'll just use the `Sentry.setXXX` methods to set data on the currently active isolation scope:
+
+```java
```

Comment:
Should we also add a Kotlin sample here?

Reply:
Yeah, let's do it in a follow up PR, I've created https://github.com/getsentry/sentry-java/issues/4067 so we don't forget.